crypto/dh/dh_group_params.c

```diff
@@ -126,15 +128,24 @@ int DH_get_nid(DH *dh)
     else
         return NID_undef;
 
-    /* Verify q is correct if it exists - reset the nid if it is not correct */
-    if (dh->params.q != NULL) {
-        BIGNUM *q = BN_dup(dh->params.p);
+    /* Calculate q = p * 2 + 1. We already know q is odd, so just shift right */
```

Comment:
The `q = p * 2 + 1` comment implies a left shift and increment.
Converting q to p would be a right shift.

Reply:
Code is right as I compared it to the known values of q in the spec.. The comment is wrong.
It is correct in the code above. RFC7919 says q = (p-1) / 2..
The wiki on safe primes says the opposite :)